### PR TITLE
BaSP-TG: Time sheet functions

### PR DESCRIPTION
**Pull request by Valentin-sanabria**
- Create timesheet
- Edit timesheet
- Delete timesheet
- Get specific timesheet
- Get all timesheets
- Filter all timesheets

### CODE REVIEW

**Review by leimeter-joaquin — CHANGES_REQUESTED** [review]

File: src/resources/time-sheets.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  }
+
+  timeSheets.splice(timeSheets.indexOf(removeThisTimesheet), 1);
+
+  res.status(200).json({
+    timeSheets,
+  });
+  fs.writeFile('./src/data/time-sheets.json', JSON.stringify(timeSheets), (err) => {
+    if (err) {
+      res.send('Created New Project');
```
Comment (kolodani) [inline src/resources/time-sheets.js:117]: el cartel de aca no coincide con el delete

**Review by kolodani — COMMENTED** [review]
vi un cartel que estaba mal valen que era el que hablamos por discord

File: src/resources/time-sheets.js (hunk 1 above)
Comment (leimeter-joaquin) [reply src/resources/time-sheets.js:117]: Yes, the message should not indicate that a proyect has been created. Be careful when copying logic form another file.

Check if this logic is correct. You are sending a success message when Json.stringify HAS an error

**Review by leimeter-joaquin — CHANGES_REQUESTED** [review]
Check this for every `fs.writeFile` in the file please

File: src/resources/time-sheets.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+const createTimeSheet = (req, res) => {
+  const newTimeSheet = req.body;
+  newTimeSheet.id = Number(new Date().getTime().toString().substring(6));
+  if (JSON.stringify(newTimeSheet) === '{}') {
+    res.status(404).send('Cannot create new Timesheet because its empty');
+  } else {
+    timeSheets.push(newTimeSheet);
+    fs.writeFile('./src/data/time-sheets.json', JSON.stringify(timeSheets), (err) => {
+      if (err) {
+        res.send('Cannot create new Project');
```
Comment (leimeter-joaquin) [inline src/resources/time-sheets.js:66]: Is there a reason you are usingn "new Project" here? If not, please change this text.

**Review by JuliBalcaza — APPROVED** [review]
Great job Valen!

**Review by JuliBalcaza — APPROVED** [review]

**Review by JuliBalcaza — APPROVED** [review]

**Review by leimeter-joaquin — APPROVED** [review]

**Pull request merged**